### PR TITLE
resolve account alias

### PR DESCRIPTION
**Pull request by mdu42**
Hi,

This PR will resolve accounts id to an alias (#27) and group iam roles by accounts. 
There is no config or independant file to specify the mapping to use. 
It is dynamically done by requesting the AWS signin page with the saml token, then parsed with BeautifulSoup. 
While making it easy to use as no specific config is required, the counter part of that method is that we are highly dependant of the aws signin page design. In that case, I've also added a command line flag (-f --noresolve) to force to switch back to the previous display.

### CODE REVIEW

**Comment by Sector95** [comment]
I commented on #27 about this, but I'm not fully understanding the need for the account number resolution.  We already collect the account information from Okta when we do the account selection, couldn't we just store that information instead?

But a step further, I'm not understanding why we need to resolve the account number for the role selection, when we already select the account the step before, or already have it defined in the profile.  I was contemplating removing the ARN from the selection, since it felt messy in the GUI anyway.

In other words, instead of `arn:aws:iam::123456789012:role/rolename`, we'd just have `rolename`.

**Comment by mdu42** [comment]
Hi,

The enhancement is useful when you have multiple aws accounts onboarded in your Okta org.

If you have  a configuration where 1 app in OKTA match 1 aws account then, I agree with you that there is no need to resolve account Id since there is no ambiguous choice as you select your app in this step.

But you can also configure OKTA in a "master/slave" configuration where you will have 1 app in OKTA that allows you to connect to all your slaves and master accounts. In that case, the app selection won't give you which account you are targetting. It also often happen that same IAM roles exists across aws accounts making it ambiguous on display.
There is two choice, either you have a strict naming convention that prevent from ambiguous names (adding your account alias in the role name for example) or you resolve accounts id.
In both situation, you may also want to group your roles with the parent aws account on display for better visibility (in our context, it often happen that an admin has more than 30 roles assigned across all accounts)

Actually that what awssignin page does (grouping and solve alias name) when you present a saml token with iam roles from multiple accounts. I prefer parsing that signin page than asking to end-users to store the mapping locally.

instead of :
```
[0] arn:aws:iam::123456789000:role/rolename
[1] arn:aws:iam::123456789001:role/rolename
[2] arn:aws:iam::123456789002:role/rolename
[3] arn:aws:iam::123456789000:role/otherrole
```
You will display :

```
Account my-account-alias-0 ( 123456789000 ) :
  [0] rolename
  [1] otherrole
Account my-account-alias-1 ( 123456789001 ) :
  [2] rolename
Account my-account-alias-2 ( 123456789002 ) :
  [3] rolename
```

**Comment by Sector95** [comment]
Got it, that makes sense.

I'd personally prefer making the alias resolution a configuration setting, and be off by default, since that's the behavior that people are expecting right now.  I know that in our environment, some of our aliases don't line up with our config profile names, so I worry we'd end up confusing people at role selection time.

"Wait, what is this account name?  I thought I selected this other account..."

We could then prompt the user, asking if they'd like alias resolution enabled during the interactive session, and store that value in the config file.

Thoughts?

**Comment by epierce** [comment]
I agree with making this off by default - in environments like ours where there are a large number of accounts with separate Okta configurations, this output is pretty confusing.  

Also, it's hardcoded for the default AWS partition. It will need to be modified to support AWS China and GovConnect - those sites use different SAML endpoints and their login pages are different as well.  I haven't looked at the HTML for those, they may have the same tags you're looking for, but they're definitely not the same page as the sign in page for all the other AWS regions.

File: gimme_aws_creds/aws.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    """
+       The Aws Client Class performes post request on AWS sign-in page
+       to fetch friendly names/alias for account and IAM roles
+    """
+
+    def __init__(self, verify_ssl_certs=True):
+        """
+        :param verify_ssl_certs: Enable/disable SSL verification
+        """
+        self._aws_signin_url = 'https://signin.aws.amazon.com/saml'
```
Comment (epierce) [inline gimme_aws_creds/aws.py]: Instead of hardcoding this value, you need to detect what AWS partition (AWS-China, GovCloud) the SAML response was generated for.  This is already being determined because it's needed by boto, so it just needs to get passed into this class.

Comment (mdu42) [reply gimme_aws_creds/aws.py]: Yes you're right. I'll change this.

**Comment by mdu42** [comment]
Indeed it can be confusing especially if the name of the app in OKTA don't lineup with aws account alias. I'll make the alias resolution optionnal and store it into the config file.

**Comment by Sector95** [comment]
Let us know when this is ready for another look.  We just merged in a change that caused a conflict, so that will need to be resolved prior to merging as well.

**Comment by mdu42** [comment]
The feature is ready. I ve just resolved the last conflict.

File: gimme_aws_creds/aws.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        for role_pair in role_pairs:
+            idp, role = None, None
+            for field in role_pair.split(','):
+                if 'saml-provider' in field:
+                    idp = field
+                elif 'role' in field:
+                    role = field
+            if not idp or not role:
+                print('Parsing error on {}'.format(role_pair))
+                exit()
```
Comment (brysontyrrell) [inline gimme_aws_creds/aws.py]: I'd like to request these `exit()` calls be updated to `sys.exit()`.

Comment (Sector95) [reply gimme_aws_creds/aws.py]: I actually had no idea there was a difference, but evidently `exit()` is considered bad practice in production code:

https://stackoverflow.com/a/19747562

There's some more code organization work that really should be done at some point as well (ie. Reworking the OktaClient class a bit to not make user input calls), maybe we roll these changes into that.

**Comment by epierce** [comment]
Looks good.  Merging the changes.
